src/v3/src/transformer/transaction/transform.ts

```diff
@@ -13,26 +13,37 @@
 import { APIError, IdxRemediation } from '@okta/okta-auth-js';
 import { flow } from 'lodash';
 
-import { IDX_STEP } from '../../constants';
+import { AUTHENTICATOR_KEY, IDX_STEP } from '../../constants';
 import { RegistrationElementSchema, TransformStepFnWithOptions } from '../../types';
 import {
   convertIdxInputsToRegistrationSchema,
   convertRegistrationSchemaToIdxInputs,
+  getAuthenticatorKey,
+  isVerifyFlow,
   parseRegistrationSchema,
   triggerRegistrationErrorMessages,
 } from '../../util';
 
 export const PIV_TYPE = 'X509';
 
 const transformRemediationNameForIdp: TransformStepFnWithOptions = (options) => (formbag) => {
-  const { transaction: { neededToProceed: remediations, nextStep } } = options;
+  const { transaction } = options;
+  const { neededToProceed: remediations, nextStep } = transaction;
   if (!remediations.length) {
     return formbag;
   }
 
+  if (nextStep?.name === IDX_STEP.REDIRECT_IDP
+    && getAuthenticatorKey(transaction) === AUTHENTICATOR_KEY.IDP) {
```

Comment:
i dont think there should be any overlap since piv cant be used as a factor for MFA, only for SSO on identify.  So there shouldnt be a case where there is a remediation with type `PIV_TYPE` with an authenticator key of `IDP`